// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`production-formatter encodes as parseable JSON 1`] = `
{
  "application": {
    "name": "One App",
    "version": "1.2.3-abc123",
  },
  "device": {
    "agent": "Darwin x64",
    "id": "host-123:1234",
  },
  "level": "error",
  "message": "1 2 3",
  "nodeFramework": "fastify",
  "schemaVersion": "0.3.0",
  "timestamp": "2018-03-02T02:39:32.948Z",
}
`;

exports[`production-formatter errors encodes ClientReportedError Error as parseable JSON 1`] = `
{
  "application": {
    "name": "One App",
    "version": "1.2.3-abc123",
  },
  "device": {
    "agent": "Browser/5.0 (compatible; WXYZ 100.0; Doors LQ 81.4; Boat/1.0)",
  },
  "error": {
    "message": "something broke",
    "name": "ClientReportedError",
    "stacktrace": "Error: something broke
    at methodA (resource-a.js:1:1)
    at methodB (resource-b.js:1:1)
",
  },
  "level": "error",
  "nodeFramework": "fastify",
  "request": {
    "address": {
      "uri": "https://example.com/page-the/error/occurred-on",
    },
    "metaData": {
      "code": "500",
      "collectionMethod": "applicationError",
      "correlationId": "123abc",
      "moduleID": "healthy-frank",
    },
  },
  "schemaVersion": "0.3.0",
  "timestamp": "2018-03-02T02:39:32.948Z",
}
`;

exports[`production-formatter errors encodes Server Reported Error as parseable JSON with nested metadata objects 1`] = `
{
  "application": {
    "name": "One App",
    "version": "1.2.3-abc123",
  },
  "device": {
    "agent": "Darwin x64",
    "id": "host-123:1234",
  },
  "error": {
    "message": "something broke",
    "name": "Error",
    "stacktrace": "Error: something broke
    at methodA (resource-a.js:1:1)
    at methodB (resource-b.js:1:1)
",
  },
  "level": "error",
  "metaData": {
    "moduleID": "healthy-frank",
    "nestedObject": {
      "level1": {
        "level2": {
          "level3": {
            "level4": "logs nested objects correctly",
          },
        },
      },
    },
  },
  "nodeFramework": "fastify",
  "schemaVersion": "0.3.0",
  "timestamp": "2018-03-02T02:39:32.948Z",
}
`;

exports[`production-formatter errors encodes as parseable JSON 1`] = `
{
  "application": {
    "name": "One App",
    "version": "1.2.3-abc123",
  },
  "device": {
    "agent": "Darwin x64",
    "id": "host-123:1234",
  },
  "error": {
    "message": "test error",
    "name": "Error",
    "stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
  },
  "level": "error",
  "message": "unable to do the thing",
  "nodeFramework": "fastify",
  "schemaVersion": "0.3.0",
  "timestamp": "2018-03-02T02:39:32.948Z",
}
`;

exports[`production-formatter errors puts a lone error in the error field 1`] = `
{
  "message": "test error",
  "name": "Error",
  "stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`production-formatter errors puts an error paired with a message in the error field 1`] = `
{
  "message": "test error",
  "name": "Error",
  "stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`production-formatter errors puts an error with other arguments in the error field and uses only the other args for the message 1`] = `
{
  "message": "test error",
  "name": "Error",
  "stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`production-formatter errors uses the error in the message when the error is paired with a message and other arguments 1`] = `
{
  "message": "test error",
  "name": "Error",
  "stacktrace": "Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7)",
}
`;

exports[`production-formatter errors uses the error in the message when the error is paired with a message and other arguments 2`] = `
"unable to do the thing Error: test error
  at Object.<anonymous> (one-app/__tests__/server/utils/logging/production-formatter.spec.js:36:64)
  at Object.asyncFn (one-app/node_modules/jest-jasmine2/build/jasmine_async.js:124:345)
  at resolve (one-app/node_modules/jest-jasmine2/build/queue_runner.js:46:12)
  at new Promise (<anonymous>)
  at mapper (one-app/node_modules/jest-jasmine2/build/queue_runner.js:34:499)
  at promise.then (one-app/node_modules/jest-jasmine2/build/queue_runner.js:74:39)
  at <anonymous>
  at process._tickCallback (internal/process/next_tick.js:188:7) 12 times"
`;

exports[`production-formatter types encodes request as parseable JSON 1`] = `
{
  "application": {
    "name": "One App",
    "version": "1.2.3-abc123",
  },
  "device": {
    "agent": "Darwin x64",
    "id": "host-123:1234",
  },
  "level": "error",
  "nodeFramework": "fastify",
  "request": {
    "address": {
      "url": "https://example.org/server",
    },
    "direction": "in",
    "metaData": {
      "correlationId": "123",
      "host": "example.org",
      "method": "GET",
      "referrer": "https://example.org/other-page",
      "userAgent": "Browser/8.0 (compatible; WXYZ 100.0; Doors LQ 81.4; Boat/1.0)",
    },
    "protocol": "https",
    "statusCode": 200,
    "statusText": "OK",
    "timings": {
      "duration": 10,
      "ttfb": 20,
    },
  },
  "schemaVersion": "0.3.0",
  "timestamp": "2018-03-02T02:39:32.948Z",
}
`;

exports[`production-formatter types uses the default util.format when given an unknown type 1`] = `
{
  "application": {
    "name": "One App",
    "version": "1.2.3-abc123",
  },
  "device": {
    "agent": "Darwin x64",
    "id": "host-123:1234",
  },
  "level": "error",
  "message": "{ type: 'yolo' }",
  "nodeFramework": "fastify",
  "schemaVersion": "0.3.0",
  "timestamp": "2018-03-02T02:39:32.948Z",
}
`;
